hibernate-core/src/test/java/org/hibernate/orm/test/metamodel/mapping/collections/SetOperationTests.java

```diff
@@ -119,8 +124,7 @@ public void testSqmFetch(SessionFactoryScope scope) {
 					).getSingleResult();
 
 					assertThat( entity.getSetOfBasics(), InitializationCheckMatcher.isInitialized() );
-
-					assert  entity.getSetOfBasics().size() == 2;
+					assertThat( entity.getSetOfBasics(), IsCollectionWithSize.hasSize( 2 ) );
```

Comment:
I think reusing existing hamcrest stuff enjoys many benefits, one of which is the detailed error message.